### PR TITLE
bpo-46765: Replace Locally Cached Strings with Statically Initialized Objects

### PR DESCRIPTION
**Pull request by ericsnowcurrently**
https://bugs.python.org/issue46765

### CODE REVIEW

File: Objects/boolobject.c [hunk 1]
```diff
@@ -1,26 +1,20 @@
...
-        s = false_str ? false_str :
-            (false_str = PyUnicode_InternFromString("False"));
-    Py_XINCREF(s);
-    return s;
+    if (self == Py_True) {
+        return &_Py_ID(True);
+    }
+    else {
+        return &_Py_ID(False);
+    }
```
Comment (arhadthedev) [inline Objects/boolobject.c]: Wouldn't ternary operator be more compact here?

```suggestion
    return (self == Py_True) ? &_Py_ID(True) : &_Py_ID(False);
```

Comment (ericsnowcurrently) [reply Objects/boolobject.c]: You bet.  Mostly I was matching the structure of the existing code.  However, I don't mind a little cleanup like this.

**Comment by ericsnowcurrently** [comment]
Alrighty then!  asyncio on Mac and refleaks...

**Comment by markshannon** [comment]
The `asycnio` test on Mac seems to be flaky. I've seen a failure on my PRs, which was fixed by re-running the tests.

The refleaks failures are puzzling. Are they passing on main?

**Comment by ericsnowcurrently** [comment]
> The `asycnio` test on Mac seems to be flaky. I've seen a failure on my PRs, which was fixed by re-running the tests.

I suspected that but wasn't sure.  It failed on repeated CI runs and on that buildbot.  So I as at least going to investigate further.
 
> The refleaks failures are puzzling. Are they passing on main?

Hmm, looks like they weren't.  So maybe I'm okay here.  I'll have to re-run it after the refleaks fix lands.

**Comment by erlend-aasland** [comment]
> Hmm, looks like they weren't. So maybe I'm okay here. I'll have to re-run it after the refleaks fix lands.

FYI, they landed with GH-31389 just recently.

**Comment by erlend-aasland** [comment]
I believe you will need to sync with `main` first:

```
$ git fetch --all
$ git merge --no-ff upstream/main
$ git push origin global-objects-use-pyid
```

**Comment by ericsnowcurrently** [comment]
That looks better.  I'm still going to double-check that macOS failure before merging.

**Comment by sweeneyde** [comment]
I tested `./python -m test test_decimal -R3:3` and got `[1, 1, 1]` reference leaks on main, and bisected that to here. Is there some reason why that could happen on my machine but not on buildbots? I tested on windows with MSVC and also on WSL.

```
1f455361ecfb1892e375bdbee813cdf095b6cfb8 is the first bad commit
commit 1f455361ecfb1892e375bdbee813cdf095b6cfb8
Author: Eric Snow <ericsnowcurrently@gmail.com>
Date:   Tue Feb 22 17:23:51 2022 -0700

    bpo-46765: Replace Locally Cached Strings with Statically Initialized Objects (gh-31366)

    https://bugs.python.org/issue46765
```

**Comment by ericsnowcurrently** [comment]
I'll take a look.

**Comment by sweeneyde** [comment]
Running `git clean -f` ended up fixing everything, so probably nothing to worry about.

**Pull request merged**